app/routes/__index/certificate/index.tsx

```diff
@@ -1,43 +1,80 @@
 import { Flex, Center } from '@chakra-ui/react';
-import type { Certificate } from '@prisma/client';
-import { useState } from 'react';
+import type { LoaderArgs, ActionArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+import { typedjson, useTypedLoaderData } from 'remix-typedjson';
+import { useRevalidator } from '@remix-run/react';
+import { useInterval } from 'react-use';
 
-import undrawSvg from '~/assets/undraw_processing_re_tbdu.svg';
+import { requireUser, requireUsername } from '~/session.server';
+import pendingSvg from '~/assets/undraw_processing_re_tbdu.svg';
 import Loading from '~/components/display-page';
 import CertificateAvailable from '~/components/certificate/certificate-available';
 import CertificateRequestView from '~/components/certificate/certificate-request';
 import { useUser } from '~/utils';
+import { getCertificateByUsername } from '~/models/certificate.server';
+import { addCertRequest } from '~/queues/certificate/certificate-flow.server';
 
-export default function CertificateIndexRoute() {
-  const user = useUser();
-  const [certificateRequested, setCertificateRequested] = useState(false);
-  const [loading, setLoading] = useState(false);
+export const loader = async ({ request }: LoaderArgs) => {
+  const username = await requireUsername(request);
+
+  try {
+    const certificate = await getCertificateByUsername(username);
+    return typedjson({ certificate });
+  } catch (error: any) {
+    throw new Error('Error retrieving certificate: ' + error.message);
```

Comment:
@Myrfion is this the right way to do the error?

Reply:
@Ririio I do errors kinda differently, here is the example:

```typescript
 throw new Response("Not Found", {
      status: 404
    });
```

Reply:
Sure I can change that. I was just wondering are we changing all error handling with `response` ?I thought they more for user interaction error like looking for a dnsRecord that doesn't exists, or routes that we don't have?

Reply:
@Ririio: Do you want to file a follow-up and fix this there?  I'd like to avoid blocking this further

Reply:
I created a follow up